internal/winapi/process.go

```diff
@@ -0,0 +1,100 @@
+package winapi
+
+import "syscall"
+
+const PROC_THREAD_ATTRIBUTE_PARENT_PROCESS = 0x00020000
+
+const PROCESS_ALL_ACCESS uint32 = 0x1FFFFF
+
+// typedef struct _PROC_THREAD_ATTRIBUTE {
+//     ULONG_PTR Attribute;
+//     SIZE_T Size;
+//     ULONG_PTR Value;
+// } PROC_THREAD_ATTRIBUTE, *PPROC_THREAD_ATTRIBUTE;
+type PROC_THREAD_ATTRIBUTE struct {
+	Attribute *uint32
+	Size      uintptr
+	Value     uintptr
+}
+
+// typedef struct _PROC_THREAD_ATTRIBUTE_LIST {
+//     ULONG PresentFlags;
+//     ULONG AttributeCount;
+//     ULONG LastAttribute;
+//     ULONG SpareUlong0;
+//     PPROC_THREAD_ATTRIBUTE ExtendedFlagsAttribute;
+//     PROC_THREAD_ATTRIBUTE Attributes[1];
+// } PROC_THREAD_ATTRIBUTE_LIST;
+type PROC_THREAD_ATTRIBUTE_LIST struct {
+	PresentFlags           uint32
+	AttributeCount         uint32
+	LastAttribute          uint32
+	SpareUlong0            uint32
+	ExtendedFlagsAttribute PROC_THREAD_ATTRIBUTE
+	Attributes             [1]PROC_THREAD_ATTRIBUTE
+}
+
+// typedef struct _STARTUPINFOEXW {
+// 		STARTUPINFOW                 StartupInfo;
+// 		LPPROC_THREAD_ATTRIBUTE_LIST lpAttributeList;
+// } STARTUPINFOEXW, *LPSTARTUPINFOEXW;
+type STARTUPINFOEX struct {
+	syscall.StartupInfo
+	LpAttributeList *PROC_THREAD_ATTRIBUTE_LIST
+}
+
+// BOOL InitializeProcThreadAttributeList(
+// 	LPPROC_THREAD_ATTRIBUTE_LIST lpAttributeList,
+// 	DWORD                        dwAttributeCount,
+// 	DWORD                        dwFlags,
+// 	PSIZE_T                      lpSize
+// );
+//
+//sys InitializeProcThreadAttributeList(list *PROC_THREAD_ATTRIBUTE_LIST, dwAttributeCount uint32, dwFlags uint32, lpSize *uintptr) (err error) = kernel32.InitializeProcThreadAttributeList
+
+// BOOL UpdateProcThreadAttribute(
+// 	LPPROC_THREAD_ATTRIBUTE_LIST lpAttributeList,
+// 	DWORD                        dwFlags,
+// 	DWORD_PTR                    Attribute,
+// 	PVOID                        lpValue,
+// 	SIZE_T                       cbSize,
+// 	PVOID                        lpPreviousValue,
+// 	PSIZE_T                      lpReturnSize
+// );
+//
+//sys UpdateProcThreadAttribute(lpAttributeList *PROC_THREAD_ATTRIBUTE_LIST, dwFlags uint32, attribute uintptr, lpValue *uintptr, cbSize uintptr, lpPreviousValue uintptr, lpReturnSize *uintptr) (err error) = kernel32.UpdateProcThreadAttribute
```

Comment:
why do lpValue and lpPreviousValue have different types in the //sys definition, but refer to the same type in the above function definition?

Reply:
Good catch, they should both be pointers iirc